mlflow/tensorflow/__init__.py

```diff
@@ -123,10 +125,10 @@ def get_default_conda_env():
 def log_model(
     model,
     artifact_path,
+    signature: ModelSignature,
```

Comment:
Why do we need to enforce `ModelSignature`?

Reply:
Do TF users always use `spark_udf`?

Reply:
If user want to run inference on a large distributed dataset, I think the spark_udf is the only way currently.
So I think we should support it correctly.
Spark can only support one dimension arrray value, without signature, we don't know which target shape we should reshape the input values as.

Reply:
Or another approach is,
we don't enforce `signature`, but when user run spark_udf on keras / tensorflow model, if no signature provided, then raise error ?

Reply:
I think documenting the limitation and raising error for multidimensional without signature is probably the best to not annoy users who aren't trying to do this

Reply:
I don't think we should enforce signature

Reply:
We should also print a warning when the signature isn't specified saying that spark_udf won't work.

Reply:
One issue: How to know the model input is multidimensional case ? (if user does not provide signature)

Reply:
Got it - in that case, no need to throw an exception for the multi dimensional case. We can Just print a warning if signature is not specified